app/jobs/alert_notify_job.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+# Job for enqueuing notifications about changes to the User account
+class AlertNotifyJob < ApplicationJob
+  queue_as :mailer
+
+  def perform(alert_id)
+    alert = Alert.find(alert_id)
+    record = alert.record
+    users = record.associated_users
+    users.each do |user|
```

Comment:
If the job failed for one user, should continue with the next one? can you make sure that this is happening?